pkg/flavor/kubernetes/flavor.go

```diff
@@ -224,12 +234,24 @@ func (flavor *Flavor) LoadNodeInfo(node *model.Node) (string, error) {
 			updateNodeRequired = true
 		}
 
+		chapUser := flavor.GetChapUserFromEnvironment()
+		if !reflect.DeepEqual(nodeInfo.Spec.ChapUser, chapUser) {
+			nodeInfo.Spec.ChapUser = chapUser
+			updateNodeRequired = true
+		}
+
+		chapPassword := flavor.GetChapPasswordFromEnvironment()
+		if !reflect.DeepEqual(nodeInfo.Spec.ChapPassword, chapPassword) {
+			nodeInfo.Spec.ChapPassword = chapPassword
```

Comment:
Do we need to update chap with encoded value like we do here https://github.com/hpe-storage/csi-driver/blob/master/pkg/flavor/kubernetes/flavor.go#L250?

Reply:
Yes, thanks for catching this.